Source/JavaScriptCore/bytecode/StructureStubInfo.h

```diff
@@ -204,28 +204,38 @@ class StructureStubInfo {
     // Not ByVal and ById case: e.g. instanceof, by-index etc.
     ALWAYS_INLINE bool considerCachingGeneric(VM& vm, CodeBlock* codeBlock, Structure* structure)
     {
-        return considerCaching(vm, codeBlock, structure, CacheableIdentifier());
+        // We never cache non-cells.
+        if (!structure) {
+            sawNonCell = true;
+            return false;
+        }
+        return considerCachingImpl(vm, codeBlock, structure, CacheableIdentifier());
     }
 
     ALWAYS_INLINE bool considerCachingBy(VM& vm, CodeBlock* codeBlock, Structure* structure, CacheableIdentifier impl)
     {
-        return considerCaching(vm, codeBlock, structure, impl);
+        // We never cache non-cells.
+        if (!structure) {
+            sawNonCell = true;
+            return false;
+        }
+        return considerCachingImpl(vm, codeBlock, structure, impl);
+    }
+
+    ALWAYS_INLINE bool considerCachingMegamorphic(VM& vm)
+    {
+        return considerCachingImpl(vm, nullptr, nullptr, CacheableIdentifier());
     }
 
     Structure* inlineAccessBaseStructure() const
     {
         return m_inlineAccessBaseStructureID.get();
     }
 private:
-    ALWAYS_INLINE bool considerCaching(VM& vm, CodeBlock* codeBlock, Structure* structure, CacheableIdentifier impl)
+    ALWAYS_INLINE bool considerCachingImpl(VM& vm, CodeBlock* codeBlock, Structure* structure, CacheableIdentifier impl)
```

Comment:
Is the intent of this function to decide to repatch? Would it be correct to rename it to considerChangingCache or considerRepatchingCache? In the megamorphic case, we should not be adding new access cases, so this is really only making sure that we don't remove the megamorphic cache on the first miss, right?

Reply:
Yes, this function is used to decide the repatching, which spmetimes making IC as generic (as before. Previously, we did make the IC generic if we see too many cases / see GiveUp case).
I think renaming is OK